lib/spack/llnl/util/filesystem.py

```diff
@@ -382,7 +382,7 @@ def unset_executable_mode(path):
     os.chmod(path, mode)
 
 
-def copy(src, dest, _permissions=False):
+def copy(src, dest, _permissions=False, symlinks=False):
```

Comment:
Would it be less confusing to users to follow the same naming convention as shutil and use `follow_symlinks` here?

Also, should be named `_symlinks` (or, `_follow_symlinks`) to avoid mixing internal and normal arguments. Alternatively, switch the order of the arguments.